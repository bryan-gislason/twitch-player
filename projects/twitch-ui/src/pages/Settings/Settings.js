import { useTranslation } from "@twitch-player/i18n";
import React, { memo } from "react";
import { View } from "react-native";
import Heading from "../../components/Heading";
import LanguagePicker from "../../components/LanguagePicker";

const Settings = memo(({ testID = "settings-page" }) => {
  const [t] = useTranslation();

  return (
    <View testID={testID}>
      <Heading level="2">{t("pages.settings.mainHeading")}</Heading>
      <LanguagePicker />
    </View>
  );
});

export default Settings;
