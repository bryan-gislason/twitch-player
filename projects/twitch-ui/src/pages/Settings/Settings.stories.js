import { storiesOf } from "@storybook/react";
import React from "react";
import Main from "../../components/Main";
import Settings from ".";

storiesOf("Pages|Settings", module).add("default", () => (
  <Main>
    <Settings />
  </Main>
));
