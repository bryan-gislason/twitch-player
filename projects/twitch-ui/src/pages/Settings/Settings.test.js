import React from "react";
import { render } from "../../testing";
import Settings from "./Settings";

jest.mock("../../components/DebugSwitch", () => "mock-debug-switch");
jest.mock("../../components/LanguagePicker", () => "mock-language-picker");
jest.mock("../../components/ThemeSwitch", () => "mock-theme-switch");

describe("Settings", () => {
  let instance;

  beforeEach(() => {
    instance = render(<Settings />);
  });

  it("renders the page", () => {
    expect(instance.getByTestId("settings-page")).toMatchSnapshot();
  });
});
