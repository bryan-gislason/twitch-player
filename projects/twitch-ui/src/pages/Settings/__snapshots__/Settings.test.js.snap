// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders the page 1`] = `
<div
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexShrink-1qe8dj5 rn-flexBasis-1mlwlqe rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="settings-page"
>
  <h2
    aria-level="2"
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-1471scf rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-vw2c0b rn-lineHeight-a8ghvy rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    data-testid="heading"
    dir="auto"
    role="heading"
    style="color: rgb(50, 47, 55); font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu, 'Helvetica Neue', sans-serif; font-size: 16px; margin: 20px 20px 5px 20px;"
  >
    Settings
  </h2>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
    style="font-size: 16px;"
  >
    <select
      class="rn-fontFamily-14xgk7a rn-fontSize-7cikom rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-width-13qz1uu"
      data-focusable="true"
      data-testid="language-picker"
      style="background-color: rgb(250, 249, 250); color: rgb(25, 23, 28);"
    >
      <option
        value="en"
      >
        English
      </option>
      <option
        value="es"
      >
        Español
      </option>
      <option
        value="fr"
      >
        Français
      </option>
      <option
        value="de"
      >
        Deutsch
      </option>
      <option
        value="it"
      >
        Italiano
      </option>
      <option
        value="he"
      >
        עִבְרִית
      </option>
      <option
        value="ar"
      >
        العَرَبِيَّة‎
      </option>
    </select>
  </div>
</div>
`;
